### ISSUE TITLE
 Error Messages for Start/End Filter

### ISSUE THREAD

**Issue by sarjona** [jleyva/moodle-block_configurablereports#113]
Now getting error messages for any reports using the Start/End filter (with the new CR version):

```
line 808 of /lib/moodlelib.php: coding_exception thrown
line 166 of /blocks/configurable_reports/report.class.php call to clearn_param()
line 80 of /blocks/configurable_reports/viewreports.php: call to report_base-->check_filters_request()

```

It has been reported by David Heuring in https://moodle.org/mod/forum/discuss.php?d=382639:

**Comment by lazarovli** [comment]
Change line 166 of /blocks/configurable_reports/report.class.php
from 
$val = clean_param($val, PARAM_CLEANHTML);
to
if (is_array($val) and is_callable('clean_param_array')) {

                                $val = clean_param_array($val, PARAM_CLEANHTML);

                        } else { 

                                $val = clean_param($val, PARAM_CLEANHTML);

                        }

**Comment by jtsafran** [comment]
The code above works for a few sites I have this on.  Just make sure to include that first IF statement that was left out of the code block :-)

```
if (is_array($val) and is_callable('clean_param_array')) {
    $val = clean_param_array($val, PARAM_CLEANHTML);
} else { 
    $val = clean_param($val, PARAM_CLEANHTML);
}
```

**Comment by sarjona** [comment]
Thanks @lazarovli and @jtsafran . It was fixed in the commit 1761a827c8b532a87c66b183be0ae448e7a1e3d1 with a similar solution you've proposed (I think the is_callable('clean_param_array') is not needed here).

I'm going to close the issue, but if the error is still present, please let me know :-)
